src/Traits/Livewire/CreatesDocuments.php

```diff
@@ -0,0 +1,234 @@
+<?php
+
+namespace FluxErp\Traits\Livewire;
+
+use FluxErp\Actions\Printing;
+use FluxErp\Contracts\OffersPrinting;
+use FluxErp\Models\Media;
+use FluxErp\View\Printing\PrintableView;
+use Illuminate\Support\Arr;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Str;
+use Illuminate\Validation\ValidationException;
+use Illuminate\View\View;
+use Laravel\SerializableClosure\SerializableClosure;
+use Livewire\Attributes\Renderless;
+use Spatie\MediaLibrary\HasMedia;
+use Spatie\MediaLibrary\Support\MediaStream;
+use Spatie\Permission\Exceptions\UnauthorizedException;
+use WireUi\Traits\Actions;
+
+trait CreatesDocuments
+{
+    use Actions, EnsureUsedInLivewire;
+
+    public array $printLayouts = [];
+
+    public array $selectedPrintLayouts = [
+        'print' => [],
+        'email' => [],
+        'download' => [],
+        'force' => [],
+    ];
+
+    public function renderCreateDocumentsModal(): View
+    {
+        return view('flux::livewire.create-documents-modal');
+    }
+
+    #[Renderless]
+    public function openCreateDocumentsModal(): void
+    {
+        $this->printLayouts = array_map(
+            fn (string $layout) => ['layout' => $layout, 'label' => __($layout)],
+            $this->getPrintLayouts()
+        );
+
+        $this->js(<<<'JS'
+            $openModal('create-documents');
+        JS);
+    }
+
+    #[Renderless]
+    protected function createDocumentFromItems(
```

Comment:
re-think what $async does. This current implementation can not be the correct way.

Reply:
√